cmd/prometheus/main.go

```diff
@@ -314,14 +298,29 @@ func main() {
 		}
 	}()
 
-	// Start all components. The order is NOT arbitrary.
-	defer func() {
-		if err := fanoutStorage.Close(); err != nil {
-			level.Error(logger).Log("msg", "Closing storage(s) failed", "err", err)
+	// Start all components while we wait for TSDB to open but only load
+	// initial config and mark ourselves as ready after it completed.
```

Comment:
Is it safe to start already the reload signal handler above?

Reply:
You mean the goroutine reading HUP signals from the channel? That can be anywhere really. Just waits for `hupReady` to get closed anyway.

Reply:
Ah, I had missed that line. :+1: